doc/schemas/lightning-getroutes.json

```diff
@@ -0,0 +1,174 @@
+    "",
+    "There are two automatic layers: *auto.localchans* contains information on local channels from this node (including non-public ones), and their exact current spendable capacities, and *auto.sourcefree* overrides all channels leading out of the *source* to be zero fee and zero delay.  These are both useful in the case where the source is the current node."
+  ],
+  "categories": [
+    "readonly"
+  ],
+  "request": {
+    "required": [
+      "source",
+      "destination",
+      "amount_msat",
+      "layers",
+      "maxfee_msat",
+      "finalcltv"
+    ],
+    "properties": {
+      "source": {
+        "type": "pubkey",
+        "description": [
+          "Node pubkey to start the payment at"
+        ]
+      },
+      "destination": {
+        "type": "pubkey",
+        "description": [
+          "Node pubkey to start the payment at"
+        ]
+      },
+      "amount_msat": {
+        "type": "msat",
+        "description": [
+          "Amount to send. It can be a whole number, or a whole number ending in *msat* or *sat*, or a number with three decimal places ending in *sat*, or a number with 1 to 11 decimal places ending in *btc*."
+        ]
+      },
+      "layers": {
+        "type": "array",
+        "items": {
+            "type": "string",
+            "description": [
+		"Layer to apply to the gossip map before attempting to find routes."
+            ]
+        }
+      },
+      "maxfee_msat": {
+        "type": "msat",
+        "description": [
+          "Maximum fee to spend: we will never return a set of routes more expensive than this. It can be a whole number, or a whole number ending in *msat* or *sat*, or a number with three decimal places ending in *sat*, or a number with 1 to 11 decimal places ending in *btc*."
+        ]
+      },
+      "finalcltv": {
+        "type": "u32",
+        "description": [
+          "Number of blocks for the final node.  We need to know this because no HTLC is allowed to have a CLTV delay more than 2016 blocks."
+        ]
+      }
+    }
+  },
+  "response": {
+    "required": [
+      "probability_ppm",
+      "routes"
+    ],
+    "properties": {
+      "probability_ppm": {
+        "type": "u64",
+        "description": [
+          "The estimated probability of success using these routes, in millionths."
+        ]
+      },
+      "routes": {
+        "type": "array",
+        "items": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": [
+            "probability_ppm",
+            "amount_msat",
+            "path"
+          ],
+          "properties": {
+            "probability_ppm": {
+              "type": "u64",
+              "description": [
+                "The estimated probability of success using this route, in millionths."
```

Comment:
I am not sure. This might be misleading.
If I give you a set of routes that fulfill the payment, eg. `rset = {r1, r2}` (the sum of the amounts in `r1` and `r2` equals the total payment amount), I can tell you the probability of `rset`, but the individual probabilities of the routes
are not necessarily independent, ie. `prob(rset) != prob(r1) * prob(r2)`. In fact `prob(r1)` means the probability of success of `r1` before I push any other route so I don't use up these channels' liquidity.

Reply:
Yes, the total probability we give takes this into account, whereas the per-path probability assumes independence (following your excellent commentary on that code).  

I'm not sure how useful these numbers will be in practice, either.  I guess you could see that one path is very unlikely, but that won't happen much?